### PR TITLE
Changes the quat_box_minus implementation and adds quat_box_plus and rigid_body_twist_transform

### PR DESCRIPTION
**Pull request by jtigue-bdai**
# Description

- Changes the quat_box_minus implementation see changes for reference links
- Adds quat_box_plus and rigid_body_twist_transform methods

# Reason for Change

While using `quat_box_minus` to get angular velocities through finite differences we encountered that the angular velocity contained some unreasonably large values at given points along the trajectory. After some investigation we observed:

- The problem comes from `quat_box_minus`  yielding unreasonably large axis-angle differences at some points (”outliers”) of the trajectory.
- Those “outliers” appear when the real part of the difference quaternion (`quat_diff = quat_mul(q1, quat_conjugate(q2))`) becomes negative ($w < 0$).
- This happens even when the inputs are standardized with $w≥0$.

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

credit to @alopez-bdai

### CODE REVIEW

File: source/isaaclab/isaaclab/utils/math.py [hunk 1]
```diff
@@ -542,6 +523,46 @@ def yaw_quat(quat: torch.Tensor) -> torch.Tensor:
...
+        q2: The second quaternion in (w, x, y, z). Shape is (N, 4).
+
+    Returns:
+        The difference between the two quaternions. Shape is (N, 3).
+
+    Reference:
+        https://github.com/ANYbotics/kindr/blob/master/doc/cheatsheet/cheatsheet_latest.pdf
+    """
+    quat_diff = quat_mul(q1, quat_conjugate(q2))  # q1 * q2^-1
+    return axis_angle_from_quat(quat_diff)  # log(qd)
```
Comment (Mayankm96) [inline source/isaaclab/isaaclab/utils/math.py:590]: Hmm was the previous implementation wrong? I remember checking that the above doesn't work well for small angles so made a better implementation.

File: source/isaaclab/isaaclab/utils/math.py [hunk 2]
```diff
@@ -781,6 +802,37 @@ def combine_frame_transforms(
     return t02, q02
 
 
+def rigid_body_twist_transform(
+    v0: torch.Tensor, w0: torch.Tensor, t01: torch.Tensor, q01: torch.Tensor
+) -> tuple[torch.Tensor, torch.Tensor]:
+    r"""Transform the linear and angular velocity of a rigid body between reference frames.
```
Comment (Mayankm96) [inline source/isaaclab/isaaclab/utils/math.py:807]: Might need to fix the docstring enters to make sure it is rendered properly.

File: source/isaaclab/isaaclab/utils/math.py [hunk 3]
```diff
@@ -499,25 +499,6 @@ def quat_mul(q1: torch.Tensor, q2: torch.Tensor) -> torch.Tensor:
...
-
-    Returns:
-        The difference between the two quaternions. Shape is (N, 3).
-    """
-    quat_diff = quat_mul(q1, quat_conjugate(q2))  # q1 * q2^-1
-    re = quat_diff[:, 0]  # real part, q = [w, x, y, z] = [re, im]
-    im = quat_diff[:, 1:]  # imaginary part
-    norm_im = torch.norm(im, dim=1)
-    scale = 2.0 * torch.where(norm_im > 1.0e-7, torch.atan2(norm_im, re) / norm_im, torch.sign(re))
-    return scale.unsqueeze(-1) * im
```
Comment (jtigue-bdai) [inline source/isaaclab/isaaclab/utils/math.py:518]: - When atan2 is being used in this function, it doesn’t account for the sign of re . Notice how the approximation takes the sign of re  to compute the scale value when norm_im  is sufficiently small but doesn’t take that into account for the non-approximated value.

- This causes that whenever there is a negative real part of the quaternion difference `re` , the scale will overestimate the rotation and will take the longest path between orientations instead of the shortest. E.g. whenever we go from -0.1 rad to 0.1 rad in a single axis this will compute a -(2pi-0.2) rad rotation instead of a 0.2 rad rotation.
- This will happen whenever `re = quat_diff[:,0]` is negative.

File: source/isaaclab/isaaclab/utils/math.py (hunk 1 above)
Comment (jtigue-bdai) [reply source/isaaclab/isaaclab/utils/math.py:590]: Hey @Mayankm96 I added some explanation to the PR description and on the removed portions of the code. Let me know if you have any questions.

File: source/isaaclab/isaaclab/utils/math.py (hunk 3 above)
Comment (jtigue-bdai) [reply source/isaaclab/isaaclab/utils/math.py:518]: @Mayankm96

File: source/isaaclab/docs/CHANGELOG.rst [hunk 4]
```diff
@@ -1,6 +1,21 @@
...
+
+Changed
+^^^^^^^
+
+* Changed the implementation of :meth:`~isaaclab.utils.math.quat_box_minus`
+
+Added
+^^^^^
+
+* Added :meth:`~isaaclab.utils.math.quat_box_minus`
```
Comment (kellyguo11) [inline source/isaaclab/docs/CHANGELOG.rst]:
```suggestion
* Added :meth:`~isaaclab.utils.math.quat_box_plus`
```

File: source/isaaclab/isaaclab/utils/math.py [hunk 5]
```diff
@@ -542,6 +523,46 @@ def yaw_quat(quat: torch.Tensor) -> torch.Tensor:
...
+
+@torch.jit.script
+def quat_box_plus(q: torch.Tensor, delta: torch.Tensor, eps: float = 1.0e-6) -> torch.Tensor:
+    """The box-plus operator (quaternion update) to apply an increment to a quaternion.
+
+    Args:
+        q: The initial quaternion in (w, x, y, z). Shape is (N, 4).
+        delta: The axis-angle perturbation. Shape is (N, 3).
+
+    eps: A small value to avoid division by zero. Defaults to 1e-6.
```
Comment (Mayankm96) [inline source/isaaclab/isaaclab/utils/math.py]: There is an enter which needs to be handled for docstrings.

```suggestion
            eps: A small value to avoid division by zero. Defaults to 1e-6.
```

File: source/isaaclab/isaaclab/utils/math.py [hunk 6]
```diff
@@ -781,6 +802,37 @@ def combine_frame_transforms(
...
+    v0: torch.Tensor, w0: torch.Tensor, t01: torch.Tensor, q01: torch.Tensor
+) -> tuple[torch.Tensor, torch.Tensor]:
+    r"""Transform the linear and angular velocity of a rigid body between reference frames.
+
+    Given the twist of 0 relative to frame 0, this function computes the twist of 1 relative to frame 1 from the position and orientation of frame 1 relative to frame 0. The transformation follows the equations:
+    .. math::
+        w_11 = R_{10} w_00 = R_{01}^{-1} w_00
+        v_11 = R_{10} v_00 + R_{10} (w_00 \times t_01) = R_{01}^{-1} (v_00 + (w_00 \times t_01))
+    where:
+        - :math:`R_{01}` is the rotation matrix from frame 0 to frame 1 derived from quaternion :math:`q_{01}`
```
Comment (Mayankm96) [inline source/isaaclab/isaaclab/utils/math.py]:
```suggestion
    Given the twist of 0 relative to frame 0, this function computes the twist of 1 relative to frame 1
    from the position and orientation of frame 1 relative to frame 0. The transformation follows the
    equations:
    
    .. math::
        w_11 = R_{10} w_00 = R_{01}^{-1} w_00
        v_11 = R_{10} v_00 + R_{10} (w_00 \times t_01) = R_{01}^{-1} (v_00 + (w_00 \times t_01))
    
    where
    
        - :math:`R_{01}` is the rotation matrix from frame 0 to frame 1 derived from quaternion :math:`q_{01}`.
```

**Review by kellyguo11 — APPROVED** [review]

**Comment by jtigue-bdai** [comment]
updated to pytest

**Comment by kellyguo11** [comment]
I think there might be something in the code that's causing a bunch of the tests to fail. Maybe `axis_angle_from_quat` needs to be moved higher up in the script

**Comment by jtigue-bdai** [comment]
> I think there might be something in the code that's causing a bunch of the tests to fail. Maybe `axis_angle_from_quat` needs to be moved higher up in the script

Yep fixed now
